tests/test_target_suitecrm.py

```diff
@@ -376,26 +392,114 @@ def test_user_delete(basic_target, suitecrm_server):
             "status": "Active",
         }
     )
+    server.create_user(
+        {
+            "user_name": "excluded",
+            "first_name": "Ernie",
+            "last_name": "Excluded",
+            "email1": "ernie.excluded@example.org",
+            "status": "Active",
+        }
+    )
 
+    remaining_user = User(
+        "adalice", forename="Ad", surname="Alice", email=("ad.alice@example.org",)
+    )
     deleted_user = User(
         "basicuser", forename="Basic", surname="Bob", email=("basic.bob@example.org",)
     )
+    excluded_user = User(
+        "excluded",
+        forename="Ernie",
+        surname="Excluded",
+        email=("ernie.excluded@example.org",),
+    )
+
+    diff = ModelDifference(
+        source_users={
+            "adAlice": remaining_user,
+            "basicuser": deleted_user,
+            "excluded": excluded_user,
+        },
+        target_users={"adAlice": remaining_user},
+        added_users={},
+        changed_users={},
+        unchanged_users={"adAlice": remaining_user},
+        removed_users={"basicuser": deleted_user, "excluded": excluded_user},
+    )
+
+    basic_target.users_cleanup(diff)
+    users = server.search_by_type("User")
+    assert users[0]["attributes"]["first_name"] == "Ad"
+    assert users[1]["attributes"]["first_name"] == "Ernie"
```

Comment:
It might not be ideal to assume the order of users is consistent. It was probably fine previously because there were only two.

```suggestion
    assert any(user["attributes"]["first_name"] == "Ad" for user in users)
    assert any(user["attributes"]["first_name"] == "Ernie" for user in users)
```